### ISSUE TITLE
Add `fn zip_longest`

### ISSUE THREAD

**Issue by TheLostLambda** [rust-itertools/itertools#962]
There is a function version of `zip_eq` already, but it appears that `zip_longest` is still only available as a method?

Let me know if you think this is a worthwhile addition for completeness's sake, and I'll submit a PR!

**Comment by Philippe-Cholet** [comment]
I don't think we stand against adding new free functions, but we sure don't intend to be complete on that side.
